utilities/fault_injection_secondary_cache.cc

```diff
@@ -88,14 +35,13 @@ std::unique_ptr<SecondaryCacheResultHandle>
 FaultInjectionSecondaryCache::Lookup(const Slice& key,
                                      const Cache::CreateCallback& create_cb,
                                      bool wait, bool& is_in_sec_cache) {
-  std::unique_ptr<SecondaryCacheResultHandle> hdl =
-      base_->Lookup(key, create_cb, wait, is_in_sec_cache);
+  std::unique_ptr<SecondaryCacheResultHandle> hdl;
   ErrorContext* ctx = GetErrorContext();
-  if (wait && ctx->rand.OneIn(prob_)) {
-    hdl.reset();
+  if (ctx->rand.OneIn(prob_)) {
+    return nullptr;
+  } else {
+    return base_->Lookup(key, create_cb, wait, is_in_sec_cache);
```

Comment:
Can we keep the FaultInjectionSecondaryCache::ResultHandle? It serves the purpose of simulating a scenario where the initial lookup returns a handle, but after the WaitAll the value is null. This is a valid scenario for the Cachelib based secondary cache.

For now, we can directly return nullptr if line 40 is true, and return FaultInjectionSecondaryCache::ResultHandle if false. We can figure out later how to properly solve the memory leak.

Reply:
This is addressed in the latest commit.